src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/ResendStaffNotificationsCoreCaseDataTest.java

```diff
@@ -0,0 +1,250 @@
+import uk.gov.hmcts.cmc.email.EmailData;
+import uk.gov.hmcts.reform.sendletter.api.SendLetterApi;
+import uk.gov.hmcts.reform.sendletter.api.SendLetterResponse;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.cmc.claimstore.repositories.CCDCaseApi.CASE_TYPE_ID;
+import static uk.gov.hmcts.cmc.claimstore.repositories.CCDCaseApi.JURISDICTION_ID;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetails;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetailsWithDefResponse;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetailsWithDefendant;
+
+@TestPropertySource(
+    properties = {
+        "document_management.api_gateway.url=false",
+        "core_case_data.api.url=http://core-case-data-api"
+    }
+)
+public class ResendStaffNotificationsCoreCaseDataTest extends BaseGetTest {
+
+    private static final String CASE_REFERENCE = "000MC023";
+
+    @MockBean
+    protected SendLetterApi sendLetterApi;
+
+    @Captor
+    private ArgumentCaptor<EmailData> emailDataArgument;
+
+    private static final String PAGE = "1";
+
+    @Before
+    public void setUp() {
+        given(pdfServiceClient.generateFromHtml(any(byte[].class), anyMap()))
+            .willReturn(new byte[]{1, 2, 3, 4});
+        UserDetails userDetails = SampleUserDetails.getDefault();
+        User user = new User(AUTHORISATION_TOKEN, userDetails);
+        given(userService.getUserDetails(AUTHORISATION_TOKEN)).willReturn(userDetails);
+        given(userService.authenticateAnonymousCaseWorker()).willReturn(user);
+        given(userService.getUser(AUTHORISATION_TOKEN)).willReturn(user);
+        given(authTokenGenerator.generate()).willReturn(SERVICE_TOKEN);
+    }
+
+    @Test
+    public void shouldRespond404WhenClaimDoesNotExist() throws Exception {
+        final String nonExistingClaimReference = "something";
+        final String event = "claim-issue";
+        given(coreCaseDataApi.searchForCitizen(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE, "page", PAGE, "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetails());
+        ResultActions result = makeRequest(nonExistingClaimReference, event);
+        assertThat(result).isNotNull();
```

Comment:
I think it should be safe to remove this assertion, testing against null is not the point of this test case and the test will still fail if null is returned. What do you think? The same applies for all other similar cases.

Reply:
Github PR codancy review moaned not having assert or fail() so it was put in place to satisfy it. Happy to change it to as long as codancy review passes that?

Reply:
We can ignore Codacy here. It tries to warn us about test cases with no assertions at all, but it does not understand Spring's `mockMvc` and does not see that we have an actual assertion in the line below. In my opinion we're good to remove :)